daemon/api_quotas_test.go

```diff
@@ -140,6 +140,46 @@ func (s *apiQuotaSuite) TestPostEnsureQuotaHappy(c *check.C) {
 	c.Assert(called, check.Equals, 1)
 }
 
+func (s *apiQuotaSuite) TestPostEnsureQuotaUpdateHappy(c *check.C) {
+	st := s.d.Overlord().State()
+	st.Lock()
+	err := servicestate.CreateQuota(st, "ginger-ale", "", nil, 9000)
+	st.Unlock()
+	c.Assert(err, check.IsNil)
+
+	r := daemon.MockServicestateCreateQuota(func(st *state.State, name string, parentName string, snaps []string, memoryLimit quantity.Size) error {
+		c.Errorf("should not have called create quota")
+		return fmt.Errorf("broken test")
+	})
+	defer r()
+
+	updateCalled := 0
+	r = daemon.MockServicestateUpdateQuota(func(st *state.State, name string, opts servicestate.QuotaGroupUpdate) error {
+		updateCalled++
+		c.Assert(name, check.Equals, "ginger-ale")
+		c.Assert(opts, check.DeepEquals, servicestate.QuotaGroupUpdate{
+			AddSnaps:       []string{"some-snap"},
+			NewMemoryLimit: 1000,
```

Comment:
should we make this larger for realism given that atm we don't support reducing memory?

Reply:
ah good point yes that is a good point